src/test/java/DocumentBuilderTest.java

```diff
@@ -0,0 +1,80 @@
+import org.junit.jupiter.api.Test;
+import org.w3c.tidy.Tidy;
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class DocumentBuilderTest {
+
+    DocumentBuilder db = new DocumentBuilder();
+
+    /**
+     * Test for the DocumentBuilder class.
+     * The first part of the test asserts that the returned HTML string from the method db.writeDoc contains
+     * the commit hash for the commits in the list.
+     *
+     * The second part of the test uses Jtidy to parse the string containing the HTML and check for errors.
+     * The line tidy.getParseErrors() == 0 will be true if getParseErrors() returns 0, meaning that no errors
+     * where found when parsing the HTML string. If getParseErrors() returns any number higher than 0 the validation
+     * has failed and there is at least one error in the HTML. Warnings are allowed and will not give any errors.
+     */
+    @Test
+    public void testWriteDoc(){
+
+        List<CommitStructure> commits = new ArrayList<>();
+        commits.add(new CommitStructure("23424324432424", "11/02-20",
+                "logloglogsfsssfsfsfsfsffs", true));
+        commits.add(new CommitStructure("79979779977979", "11/02-20",
+                "logloglogsfsssf645646465466sfsffs", true));
+
+        //set system out to an outputstream so it does not print anything when
+        //running tidy.parse(System.in, System.out)
+        System.setOut(new PrintStream(new OutputStream() {
```

Comment:
Interesting, and I see the point in this, but doesn't it affect subsequent tests as well unless the I/O is reset later? It seems the empty stream could be placed directly in the call to `parse` instead

Reply:
Yes I thought of that too. I'll look into it